### ISSUE TITLE
Rule Request: `Bool` variable naming

### ISSUE THREAD

**Issue by sjavora** [realm/SwiftLint#5034]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

It is a convention to prefix `Bool` variables with a verb to indicate the type - usually `is`/`are`, but there could be others. Examples include [UIView.isHidden](https://developer.apple.com/documentation/uikit/uiview/1622585-ishidden), [UIDeviceOrientation.isPortrait](https://developer.apple.com/documentation/uikit/uideviceorientation/1620057-isportrait) or the standard library's [Collection.isEmpty](https://developer.apple.com/documentation/swift/collection/isempty-35uy0).

Triggering examples:
```swift
var flipped = false

var changes = true

let empty: Bool
```

Non-triggering examples:
```swift
var isFlipped = false

var hasChanges = true

let isEmpty: Bool
```

The rule should have a list of prefixes that are allowed by default and the ability to specify more. From my experience, I'd start with `is`, `are`, `has`, `should`, `can`, `will`, `did`.

The rule should probably be opt-in.

I'm posting this first in case there are any red flags - I don't want to spend time developing this only to be told later that the rule won't be merged 🙂

**Comment by sjavora** [comment]
@jpsim could you please comment?

**Comment by SimplyDanny** [comment]
Sounds like a useful rule to me. However, we need to clarify if the rule applies to all or a subset of the following cases:
* Properties
* Computed properties
* Local variable declarations
* Functions returning `Bool`
* Methods returning `Bool`
